hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -157,28 +171,29 @@ SqlResult execute(DropMappingPlan plan) {
     }
 
     SqlResult execute(CreateDataLinkPlan plan) {
-        InternalDataLinkService dlService = getNodeEngine(hazelcastInstance).getDataLinkService();
-        if (plan.ifNotExists() && dlService.existsDataLink(plan.name())) {
-            throw new HazelcastException("Data link '" + plan.name() + "' already exists");
-        }
+        InternalDataLinkService dlService = nodeEngine.getDataLinkService();
+        assert !plan.ifNotExists() || !plan.isReplace();
```

Comment:
This is not a validation, we validate in `SqlCreateDataLink.validate`. This is an assertion where we assert that it's the case. It's more for the benefit of the reader to see that this is the case. Or if the original validation gets broken. For example one of the tests used `true` for both.

Reply:
if you knew, that `CreateDataLinkPlan` enforces its preconditions/invariants, you wouldn't need additional asserts because you would be sure that you will get no garbage in `CreateDataLinkPlan` object

Reply:
this assert checks invariant of the `CreateDataLinkPlan`, so I still think it is redundant. `CreateDataLinkPlan` is not a bag of random values, but a "plan" in our domain, which cannot contain any combination of values because of the way the domain is structured.

Assert might make sense if it asserted that some valid `CreateDataLinkPlan` should not be passed here because it is not supported, makes no sense, etc. But this is not the case here.